src/dynamic_graph/dynamic-graph-py.cc

```diff
@@ -0,0 +1,131 @@
+// Copyright 2010, Florent Lamiraux, Thomas Moulard, LAAS-CNRS.
+
+#include <iostream>
+#include <sstream>
+
+#include <dynamic-graph/debug.h>
+#include <dynamic-graph/exception-factory.h>
+#include <dynamic-graph/signal-base.h>
+
+#include "dynamic-graph/python/exception.hh"
+#include "dynamic-graph/python/dynamic-graph-py.hh"
+#include "dynamic-graph/python/signal-wrapper.hh"
+
+namespace dynamicgraph {
+namespace python {
+
+/**
+   \brief plug a signal into another one.
+*/
+PyObject* plug(PyObject* /*self*/, PyObject* args) {
+  PyObject* objOut = NULL;
+  PyObject* objIn = NULL;
+  void* pObjOut;
+  void* pObjIn;
+
+  if (!PyArg_ParseTuple(args, "OO", &objOut, &objIn)) return NULL;
+
+  if (!PyCapsule_CheckExact(objOut)) {
+    PyErr_SetString(PyExc_TypeError,
+                    "first argument should be a pointer to"
+                    " signalBase<int>.");
+    return NULL;
+  }
+  if (!PyCapsule_CheckExact(objIn)) {
+    PyErr_SetString(PyExc_TypeError,
+                    "second argument should be a pointer to"
+                    " signalBase<int>.");
+    return NULL;
+  }
+
+  pObjIn = PyCapsule_GetPointer(objIn, "dynamic_graph.Signal");
+  SignalBase<int>* signalIn = (SignalBase<int>*)pObjIn;
```

Comment:
Could you add 

    if (signalIn == NULL) return NULL;

With the current release, one get a very annoying SEGV when passing the wrong object.

Reply:
PyCapsule_GetPointer is responsible for setting the Python error. No need to handle it.

Reply:
I should do that, thanks for the review. I'll also provide a test to check what happens if we pass a wrong object.